packages/odyssey-react/src/components/Select/Select.module.scss

```diff
@@ -446,8 +435,10 @@
   }
 
   .focused & {
-    @include outline($focus-ring-primary);
-
+    outline-color: var(--FocusOutlineColor);
+    outline-offset: var(--FocusOutlineOffset);
+    outline-style: var(--FocusOutlineStyle);
+    outline-width: var(--FocusOutlineWidth);
```

Comment:
this works - but we'll also be duplicating styles within each focusable component and making our future maintenance a bigger pain as we don't have a single location where this logic lives. One solution would be to create a FocusRing or Focusable or similar component. We can then ship a single set of styles to the browser reducing bytes over the wire and making our UI efficient. We can also update styles for focus easily in one single place. Downstream teams will also be able to easily leverage this component to build their own focusable UI.

Some drawbacks would include the need for an intermediate DOM element and using focus-within which may not have full browser support without polyfill solutions. Thoughts?

Reply:
Agreed re: the benefits of abstraction here. Copy/pasting this over and over was not fun 😂 

Rather than make a component for it, I think it'd behoove us to figure out how we want to handle abstractions at the style level.

Today, we handle that with Sass mixins. Since we're going to rely on other mixins for the time being, I could update our existing `outline` mixin to use these styles. If we want to move away from Sass where possible, `composes` would be a CSS Modules-native solution. Finally, I *believe* CSS Modules also support imports from CSS-in-JS if we wanted to do something bespoke. I'd prefer to avoid that if we can use an existing solve, though.

Reply:
I think that getting this new focus state implemented and moving us away from the previous sass include is a big enough scope for this particular story.

There are too many unknowns and too much risk to try to optimize things further in this PR. Let's save that for its own dedicated story.

Reply:
@andrewberg-okta I've created [OKTA-475961](https://oktainc.atlassian.net/browse/OKTA-475961) to account for this spike, should we want to proceed.